src/include/hip/rpp_hip_common.hpp

```diff
@@ -72,4 +114,441 @@ inline RppStatus generate_gaussian_kernel_gpu(Rpp32f stdDev, Rpp32f* kernel, Rpp
+{
+    src_f24->x.x.x = __half2float(srcPtr[srcIdx]);
+    src_f24->x.x.y = __half2float(srcPtr[srcIdx + 3]);
+    src_f24->x.x.z = __half2float(srcPtr[srcIdx + 6]);
+    src_f24->x.x.w = __half2float(srcPtr[srcIdx + 9]);
+    src_f24->x.y.x = __half2float(srcPtr[srcIdx + 12]);
+    src_f24->x.y.y = __half2float(srcPtr[srcIdx + 15]);
+    src_f24->x.y.z = __half2float(srcPtr[srcIdx + 18]);
+    src_f24->x.y.w = __half2float(srcPtr[srcIdx + 21]);
+
+    src_f24->y.x.x = __half2float(srcPtr[srcIdx + 1]);
+    src_f24->y.x.y = __half2float(srcPtr[srcIdx + 4]);
+    src_f24->y.x.z = __half2float(srcPtr[srcIdx + 7]);
+    src_f24->y.x.w = __half2float(srcPtr[srcIdx + 10]);
+    src_f24->y.y.x = __half2float(srcPtr[srcIdx + 13]);
+    src_f24->y.y.y = __half2float(srcPtr[srcIdx + 16]);
+    src_f24->y.y.z = __half2float(srcPtr[srcIdx + 19]);
+    src_f24->y.y.w = __half2float(srcPtr[srcIdx + 22]);
+
+    src_f24->z.x.x = __half2float(srcPtr[srcIdx + 2]);
+    src_f24->z.x.y = __half2float(srcPtr[srcIdx + 5]);
+    src_f24->z.x.z = __half2float(srcPtr[srcIdx + 8]);
+    src_f24->z.x.w = __half2float(srcPtr[srcIdx + 11]);
+    src_f24->z.y.x = __half2float(srcPtr[srcIdx + 14]);
+    src_f24->z.y.y = __half2float(srcPtr[srcIdx + 17]);
+    src_f24->z.y.z = __half2float(srcPtr[srcIdx + 20]);
+    src_f24->z.y.w = __half2float(srcPtr[srcIdx + 23]);
+}
+
+// U8 loads with layout toggle PLN3 to PKD3 (24 U8 pixels)
+
+__device__ __forceinline__ void rpp_hip_load24_pln3_and_unpack_to_float24_pkd3(uchar *srcPtr, uint srcIdx, uint increment, d_float24 *src_f24)
+{
+    d_uint6 src;
+
+    src.x = *((uint2 *)(&srcPtr[srcIdx]));
+    srcIdx += increment;
+    src.y = *((uint2 *)(&srcPtr[srcIdx]));
+    srcIdx += increment;
+    src.z = *((uint2 *)(&srcPtr[srcIdx]));
+
+    src_f24->x.x = make_float4(rpp_hip_unpack0(src.x.x), rpp_hip_unpack0(src.y.x), rpp_hip_unpack0(src.z.x), rpp_hip_unpack1(src.x.x));
+    src_f24->x.y = make_float4(rpp_hip_unpack1(src.y.x), rpp_hip_unpack1(src.z.x), rpp_hip_unpack2(src.x.x), rpp_hip_unpack2(src.y.x));
+    src_f24->y.x = make_float4(rpp_hip_unpack2(src.z.x), rpp_hip_unpack3(src.x.x), rpp_hip_unpack3(src.y.x), rpp_hip_unpack3(src.z.x));
+    src_f24->y.y = make_float4(rpp_hip_unpack0(src.x.y), rpp_hip_unpack0(src.y.y), rpp_hip_unpack0(src.z.y), rpp_hip_unpack1(src.x.y));
+    src_f24->z.x = make_float4(rpp_hip_unpack1(src.y.y), rpp_hip_unpack1(src.z.y), rpp_hip_unpack2(src.x.y), rpp_hip_unpack2(src.y.y));
+    src_f24->z.y = make_float4(rpp_hip_unpack2(src.z.y), rpp_hip_unpack3(src.x.y), rpp_hip_unpack3(src.y.y), rpp_hip_unpack3(src.z.y));
+}
+
+// F32 loads with layout toggle PLN3 to PKD3 (24 F32 pixels)
+
+__device__ __forceinline__ void rpp_hip_load24_pln3_and_unpack_to_float24_pkd3(float *srcPtr, uint srcIdx, uint increment, d_float24 *src_f24)
+{
+    float *srcPtrR, *srcPtrG, *srcPtrB;
+    srcPtrR = srcPtr + srcIdx;
+    srcPtrG = srcPtrR + increment;
+    srcPtrB = srcPtrG + increment;
+
+    src_f24->x.x.x = *srcPtrR;
```

Comment:
use vector float4 for reading srcPtr memory and use its components when filling the src_f24.

Reply:
Done